File: Lunchtime/View/LunchtimeTableViewCell.m
//
//  LunchtimeTableViewCell.m
//  Lunchtime
//
//  Created by Willow Belle on 2015-11-16.
//  Copyright © 2015 Cosmic Labs. All rights reserved.
//

#import "LunchtimeTableViewCell.h"
#import "UIColor+Lunchtime.h"

@implementation LunchtimeTableViewCell

- (void)setHighlighted:(BOOL)highlighted {
    [super setHighlighted:highlighted];

    if (highlighted) {
        self.backgroundColor = [UIColor lightGray];
    } else {
        [UIView animateWithDuration:0.2 animations:^{
            self.backgroundColor = [UIColor clearColor];
        }];
    }
}

@end

File: Lunchtime/View/LunchtimeTableViewCell.swift
//
//  LunchtimeTableViewCell.swift
//  Lunchtime
//
//  Created by Willow Bumby on 2015-11-18.
//  Copyright © 2015 Lighthouse Labs. All rights reserved.
//

import UIKit

class LunchtimeTableViewCell: UITableViewCell {

    override func setHighlighted(highlighted: Bool, animated: Bool) {
        super.setHighlighted(highlighted, animated: animated)

        if highlighted {
            backgroundColor = .lightGray()
        } else {
            UIView.animateWithDuration(0.2) {
                self.backgroundColor = .clearColor()
            }
        }
    }
}